external-providers/java-external-provider/pkg/java_external_provider/util.go

```diff
@@ -553,6 +553,8 @@ func toDependency(_ context.Context, log logr.Logger, depToLabels map[string]*de
 	return dep, err
 }
 
+var mavenSearchErrorCache error
```

Comment:
Add this as something to take care of in a rewrite. I think this will be fine for now but I kind of agree with Code Rabbit and that we might want to do something different in the future